### PR TITLE
Add Terraform Scripts To Support Shared Infrastructure

### PR DESCRIPTION
**Pull request by zachclark-ccpo**
Created Terraform scripts for shared infrastructure on Azure.
The following resources are created with this implementation:

- Resource group "poc-shared-infrastructure"
- Storage account "pocsharedstorage630"
- Key vault (Standard SKU) "atat-test-1-keyvault"
- 2 Secrets to populate Key vault
- 2 RSA keys to populate Key vault

When you clone this branch for review, make sure to run `terraform init`. 
Refer to the README for additional instructions.

### CODE REVIEW

**Comment by anhtran-ccpo** [comment]
Can you rebase with develop?

**Review by anhtran-ccpo — APPROVED** [review]
Ran the commands described in the README with the tfvars supplied to me and was a success.

**Comment by zachclark-ccpo** [comment]
There are currently 2 problems:

- You cannot run `terraform apply` if the infrastructure was already deployed remotely by someone else.
- Creating the keyvault locks access to anyone who isn't Tom, Jeff, or the person who ran `terraform apply`.

**Comment by jasonburkert-ccpo** [comment]
@zachclark-ccpo Why not locked to Tom and Jeff?  Is it due to some role or membership in a group?

**Review by jasonburkert-ccpo — COMMENTED** [review]
Some minor suggestions.  Waiting to see outcome of the revision.  Hit the first problem @zachclark-ccpo mentioned.

**Review by kylelaker-ccpo — COMMENTED** [review]
Can we make sure to run `terraform fmt poc/shared_infrastructure/`

File: poc/shared_infrastructure/main.tf [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      version = ">= 2.26"
+    }
+  }
+
+  required_version = ">= 0.14.9"
+}
+
+provider "azurerm" {
+  features {}
+}
```
[the remark this answers is not in the document]
Comment (zachclark-ccpo) [reply poc/shared_infrastructure/main.tf]: Understandable, I'll update the versioning.

File: poc/shared_infrastructure/variables-auth.tf [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+variable "azure-client-secret" {
+  type        = string
+  description = "Azure Client Secret"
+}
+
+variable "azure-tenant-id" {
+  type        = string
+  description = "Azure Tenant ID"
+}
```
[the remark this answers is not in the document]
Comment (zachclark-ccpo) [reply poc/shared_infrastructure/variables-auth.tf:24]: Updated the versions in 618e7aa

File: poc/shared_infrastructure/main.tf (hunk 1 above)
[the remark this answers is not in the document]
Comment (zachclark-ccpo) [reply poc/shared_infrastructure/main.tf]: Updated config versioning 618e7aa

**Comment by zachclark-ccpo** [comment]
The 2 problems have been resolved.

File: poc/shared_infrastructure/main.tf [hunk 3]
```diff
@@ -0,0 +1,82 @@
+# Configure the Azure provider
+terraform {
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = ">= 2.65, <3.0.0"
+    }
+  }
+
+  required_version = ">= 0.10"
```
Comment (kylelaker-ccpo) [inline poc/shared_infrastructure/main.tf:10]:
```suggestion
  required_version = ">= 1.0"
```

**Review by kylelaker-ccpo — CHANGES_REQUESTED** [review]

File: poc/shared_infrastructure/main.tf (hunk 3 above)
Comment (zachclark-ccpo) [reply poc/shared_infrastructure/main.tf]: Understandable, thank you.

**Review by kylelaker-ccpo — APPROVED** [review]

**Pull request merged**